### ISSUE TITLE
Space-padded days in from_format

### ISSUE THREAD

**Issue by hynek** [sdispater/pendulum#269]
First off, thanks for this great library – I’m basically using it in all of my projects. Now I’ve run into a format oddity that seems to be more common than I thought: space-padded days.

This is used by `ls`, OpenSSL, and apparently syslog.

For example what I needed to parse today is `Apr  2 12:00:00 2020 GMT` (note the two spaces before the first 2) which requires a different from_format format than say `Apr 12 12:00:00 2020 GMT`. 

It looks like datetime's strptime [handles this](https://twitter.com/moreati/status/969329950925967361) out of the box.  True to pendulum's principles I would of course prefer an explicit way of saying “space-padded day” instead of guessing.

**Comment by sdispater** [comment]
Thanks for the kind words! I really appreciate it.

I did not know space-padded days existed, so I learned something new today.

We could take the same approach as `strptime()` and make `DD` also handle space padded days (and not only zero-padded days). So, the string could be matched by `MMM DD HH:mm:ss YYYY z`.

I don't think this particular case warrants a new token so using `DD` seems good to me.
